agent/consul/acl.go

```diff
@@ -1702,6 +1702,36 @@ func (f *aclFilter) filterServiceList(services *structs.ServiceList) {
 	*services = ret
 }
 
+// filterGatewayServices is used to filter gateway to service mappings based on ACL rules.
+func (f *aclFilter) filterGatewayServices(mappings *structs.GatewayServices) {
+	svcs := *mappings
+	var authzContext acl.AuthorizerContext
+
+	for i := 0; i < len(svcs); i++ {
+		s := svcs[i]
+
+		var drop bool
+		s.Gateway.FillAuthzContext(&authzContext)
+		if !f.allowService(s.Gateway.ID, &authzContext) {
+			drop = true
+			f.logger.Debug("dropping gateway from result due to ACLs", "gateway", s.Gateway.ID)
+		}
+
+		s.Service.FillAuthzContext(&authzContext)
+		if !f.allowService(s.Service.ID, &authzContext) {
```

Comment:
```suggestion
		if !f.allowService(s.Service.ID, &authzContext) && !drop {
```
If something triggers both conditionals, you probably only want one of them to debug log.

Reply:
Actually, why not just `continue` from each conditional directly?

Reply:
Ended up redoing this bit and pulling out the gateway read check to the RPC endpoint

Reply:
I believe the only endpoints that filter ACLs at the HTTP layer are agent endpoints (because there's no corresponding RPC endpoint).

Since this data is coming from the state store, I don't think there would be a code path requiring ACL filtering that would bypass the RPC endpoint.

Reply:
Seems reasonable to me. 👍